packages/fiori/src/IllustratedMessage.js

```diff
@@ -386,15 +387,45 @@ class IllustratedMessage extends UI5Element {
 			return;
 		}
 
+		this._applyMedia();
+	}
+
+	_applyMedia() {
+		const bcr = this.getBoundingClientRect().toJSON();
+		let newMedia = "";
+
 		if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.BASE) {
-			this.media = IllustratedMessage.MEDIA.BASE;
+			newMedia = IllustratedMessage.MEDIA.BASE;
 		} else if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.SPOT) {
-			this.media = IllustratedMessage.MEDIA.SPOT;
+			newMedia = IllustratedMessage.MEDIA.SPOT;
 		} else if (this.offsetWidth <= IllustratedMessage.BREAKPOINTS.DIALOG) {
-			this.media = IllustratedMessage.MEDIA.DIALOG;
+			newMedia = IllustratedMessage.MEDIA.DIALOG;
 		} else {
-			this.media = IllustratedMessage.MEDIA.SCENE;
+			newMedia = IllustratedMessage.MEDIA.SCENE;
+		}
+
+		if (!(this._lastKnownBCRForMedia[newMedia] && this._shallowEqual(bcr, this._lastKnownBCRForMedia[newMedia]))) {
```

Comment:
I suggest declaring a variable `const lastKnowBCR = this._lastKnownBCRForMedia[newMedia]` to avoid repetition and make the `if` statement shorter, easier to read